### PR TITLE
Jerry/add magma sgels

### PR DESCRIPTION
**Pull request by JerryChen97**

### CODE REVIEW

**Comment by Roger-luo** [comment]
@JerryChen97 gels should have a gpu interface in MAGMA right? why this only binds the CPU?

File: deps/build_OpenBLAS.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
```
Comment (Roger-luo) [inline deps/build_OpenBLAS.jl]: if this PR is about `gels`, why there is draft build script here?

Comment (JerryChen97) [reply deps/build_OpenBLAS.jl]: > @JerryChen97 gels should have a gpu interface in MAGMA right? why this only binds the CPU?

I am still trying to solve a bug related with the sgels' gpu interface in MAGMA.

Comment (JerryChen97) [reply deps/build_OpenBLAS.jl]: > if this PR is about `gels`, why there is draft build script here?

I forgot to delete them...

Comment (Roger-luo) [reply deps/build_OpenBLAS.jl]: So this PR is not ready for review yet? OK, it'd be better to add WIP in the title if this is not ready for review.

**Comment by JerryChen97** [comment]
@Roger-luo Now the bugs should have disappeared and some redundancies are cleaned. Please check and merge if available.

**Comment by JerryChen97** [comment]
Later I will go on with several other linear system solvers' wrappers

**Comment by Roger-luo** [comment]
looks good!

**Pull request merged**